docs/node/run-your-node/keymanager-node/key-manager-upgrade.md

```diff
@@ -0,0 +1,103 @@
+key material can not be restored on another machine. 
+
+**During a Key Manager node upgrade it is therefore essential that the key 
+material is not lost, not even due to an operational error or even a 
+catastrophically failed upgrade.**
+
+## Safe Upgrade Procedure
+
+A Key Manager node's upgrade procedure differs from other Oasis nodes upgrades 
+because the upgraded node cannot unseal/decrypt the old Key Manager's state.
+
+To upgrade a Key Manager node, we need to delete the local state and let the 
+Key Manager's state replicate itself from other nodes. Only one Key Manager 
+runtime in the configuration file can be present at once. 
+
+**Always follow the safe upgrade procedure:**
+1. Keep the first half of nodes running on the old version
+2. Upgrade the second half
+3. Wait for the paratime upgrade epoch
+4. Verify that secrets have been replicated to the second half of nodes
+5. **Verify again**
+6. Upgrade the first half of the nodes
+
+### Upgrade Nodes
+
+To upgrade a Key Manager, follow the next steps:
+1. Wipe its local state `worker-local-storage.badger.db`
+```
+rm -rf runtimes/4000000000000000000000000000000000000000000000004a1a53dff2ae482d/worker-local-storage.badger.db/
+```
+2. Upgrade the Key Manager runtime
+3. Wait for the key material to get replicated from active nodes before 
+continuing
+
+### After the Upgrade
+
+#### If everything goes well
+
+After the upgrade epoch and when the key material is successfully replicated, 
+the `control status` output should show `keymanager.status="ready"` and 
+`registration.descriptor.runtimes.0.extra_info` should contain a hash of the 
+key material state:
+```
+$ oasis_node oasis-node -a unix:/storage/node/internal.sock control status
+...
+  "registration": {
+    "last_registration": "2023-02-06T08:40:30Z",
+    "descriptor": {
+...
+      "runtimes": [
+        {
+          "id": "4000000000000000000000000000000000000000000000004a1a53dff2ae482d",
+          "version": {
+            "minor": 3,
+            "patch": 3
+          },
+          "capabilities": {
+            "tee": {
+              "hardware": 1,
+...
+            }
+          },
+          "extra_info": "omlzaWduYXR1cmVYQG7nDuKTOUKAlJAfukdY6Xljox376lCLI0cIP0zPw2B8abJxa31j+NoQAWA0KZuHD41XPyICmjXDTpjDXukEEgVtaW5pdF9yZXNwb25zZaNoY2hlY2tzdW1YIEWZF5YaFQChstrZ9u1UdgyqZCagmNfghvyQna9WkmvyaWlzX3NlY3VyZfVvcG9saWN5X2NoZWNrc3VtWCCsrqRzYjx05t+KoCYz7wFSdKJ720g2LQBAsRKXmClMvw=="
+        }
+      ],
+      "roles": "key-manager",
+    }
+  }
+...
+  "keymanager": {
+    "status": "ready",
+    "may_generate": false,
+    "runtime_id": "4000000000000000000000000000000000000000000000004a1a53dff2ae482d",
+    "client_runtimes": [
+      "000000000000000000000000000000000000000000000000a6d1e3ebf60dff6c",
+      "0000000000000000000000000000000000000000000000000000000000000000"
+    ],
+    "access_list": [
+...
+    ],
+    "private_peers": [
+...
+    ]
+  }
+```
+
+#### Errors
```

Comment:
Move this to a top-level "Troubleshooting" section, similar to what we have for the "ParaTime Node" page.